File: Perl/all_whales.pl
#!/usr/bin/perl -w

chomp(@lines = <STDIN>);

foreach $var (@lines){
	if($var =~ m/^\s*$/){
		next;
	}
	$var =~ tr/[A-Z]/[a-z]/;
	my $old_whale = $var;	
	$var =~ s/^[0-9 ]+ //i;
	if($var =~ m/s$/){
		$var =~ s/s$//;
	}
	$var =~ s/\s+/ /;
	$var =~ s/\s*$//;
#	print "--whale name is: $var\n";
	$old_whale =~ s/[^0-9]//gi;
#	print "++whale name indi nb is : $old_whale\n";
	if(exists($data{$var})){
		$data{$var}[0]++;
		$data{$var}[1] += $old_whale;	
	}else{
		$data{$var}[0] = 1;
		$data{$var}[1] = $old_whale;
	}
}


@whale_names = keys %data;
@whale_names = sort @whale_names;
foreach $name (@whale_names){
#	print "$name\n";
#	print "$data{$name}[0]\n";
#	print "$data{$name}[1]\n";
	print "$name observations: $data{$name}[0] pods, $data{$name}[1] individuals\n";
}

File: Python/all_whales.py
#!/usr/bin/python3

import sys
import re
from collections import defaultdict

record = sys.stdin.readlines()
dic = defaultdict(list)

for r in record:
    r = r.lower()
    r = re.sub(r's$', '', r)
    r = re.sub(r'\s+', ' ', r)
    lst = r.split()
    whale_name = ' '.join(lst[1: ])
    nb_of_individuals = int(lst[0])
   
    if whale_name not in dic:
        dic[whale_name].append(1)
        dic[whale_name].append(nb_of_individuals)
    else:
        dic[whale_name][0] += 1
        dic[whale_name][1] += nb_of_individuals
     




for key in sorted(dic):
    print('{} observations:'.format(key),
          '{} pods,'.format(dic[key][0]),
          '{} individuals'.format(dic[key][1]))


